### PR TITLE
Fix nullannotations in ItemResource and possible NPEs in ItemDTOMapper

### PR DESCRIPTION
**Pull request by triller-telekom**
Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResource.java [hunk 1]
```diff
@@ -494,7 +501,7 @@ public Response removeTag(@PathParam("itemname") @ApiParam(value = "item name",
     public Response createOrUpdateItem(
             @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(value = "language") String language,
             @PathParam("itemname") @ApiParam(value = "item name", required = true) String itemname,
-            @ApiParam(value = "item data", required = true) GroupItemDTO item) {
+            @ApiParam(value = "item data", required = true) @Nullable GroupItemDTO item) {
```
Comment (sjka) [inline bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResource.java]: How can it be `required = true` and `@Nullable` at the same time?

File: bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResource.java [hunk 2]
```diff
@@ -539,7 +546,8 @@ public Response createOrUpdateItem(
     @ApiOperation(value = "Adds a list of items to the registry or updates the existing items.")
     @ApiResponses(value = { @ApiResponse(code = 200, message = "OK", response = String.class),
             @ApiResponse(code = 400, message = "Item list is null.") })
-    public Response createOrUpdateItems(@ApiParam(value = "array of item data", required = true) GroupItemDTO[] items) {
+    public Response createOrUpdateItems(
+            @ApiParam(value = "array of item data", required = true) GroupItemDTO @Nullable [] items) {
```
Comment (sjka) [inline bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResource.java]: Again, `required = true` and `@Nullable` are contradicting each other, aren't they?

**Comment by triller-telekom** [comment]
I have removed the two `@Nullable` from the REST API interface and kept the `null` checks since this is an external interface where we cannot be sure to receive a non-null object.

**Comment by maggu2810** [comment]
I triggered Travis CI again.

**Comment by triller-telekom** [comment]
close -> reopen because of wemo test failure:

```
WemoLightHandlerOSGiTest
```

**Pull request merged**